app/controllers/checkin_controller.rb

```diff
@@ -3,13 +3,13 @@ class CheckinController < ApplicationController
 
   def new
     flash.clear
-    @user = Student.find_by_id(session[:current_user_id])
-    @checkin = Checkin.new
   end
 
   def create
     flash.clear
     Time.zone = 'Pacific Time (US & Canada)'
+    @user = Student.find_by_id(session[:current_user_id])
+    @checkin = Checkin.new
     @checkin.update(time: DateTime.now, student_id: @user.id, reason: params[:checkin][:reason])
```

Comment:
line 13: It should be `params[:reason]`, not `params[:checkin][:reason]`
![image](https://user-images.githubusercontent.com/56783069/160330021-7bb8f53f-a826-4575-a071-226ece77f3cc.png)

Reply:
I tried this while testing in development and it didn't work for me. I think there might be a difference in how params is set when the actual form is submitted (it sets :checkin => { :reason => "something" } in the request header) though you can double check this by removing the before action and changing the code in the controller to test. lmk if i'm getting this wrong